### PR TITLE
MINOR: Reduce number of threads created for integration test brokers

### PR DESCRIPTION
**Pull request by splett2**
The integration tests seem to create an unnecessarily large number of threads. This reduces the number of threads created per integration test harness broker.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/test/TestUtils.java [hunk 1]
```diff
@@ -150,17 +150,6 @@ public static String randomString(final int len) {
...
-
-        // Note that we don't use Exit.addShutdownHook here because it allows for the possibility of accidently
-        // overriding the behaviour of this hook leading to leaked files.
-        Runtime.getRuntime().addShutdownHook(KafkaThread.nonDaemon("delete-temp-file-shutdown-hook", () -> {
-            try {
-                Utils.delete(file);
-            } catch (IOException e) {
-                log.error("Error deleting {}", file.getAbsolutePath(), e);
-            }
-        }));
```
Comment (splett2) [inline clients/src/test/java/org/apache/kafka/test/TestUtils.java:162]: For temp files this extra thread does not seem necessary. `file.deleteOnExit` already registers a shutdown hook.

**Review by showuon — APPROVED** [review]
This change makes sense to me. Out of curiosity, have you noticed any test failure caused by the threads number?

**Comment by splett2** [comment]
@showuon 
not explicitly. I was hoping it would reduce some of the CI job resource utilization.

**Comment by splett2** [comment]
Test failures were:
```
[Test Result](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13655/3/testReport/) (2 failures )
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.controller.QuorumControllerTest.testUnregisterBroker()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13655/3/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_8_and_Scala_2_12___testUnregisterBroker__/)
[Build / JDK 17 and Scala 2.13 / kafka.api.TransactionsTest.testBumpTransactionalEpoch(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13655/3/testReport/junit/kafka.api/TransactionsTest/Build___JDK_17_and_Scala_2_13___testBumpTransactionalEpoch_String__quorum_kraft/)
```
I think they should be unrelated.

File: core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala [hunk 2]
```diff
@@ -609,7 +609,10 @@ class DynamicBrokerConfigTest {
     val zkClient: KafkaZkClient = mock(classOf[KafkaZkClient])
     when(zkClient.getEntityConfigs(anyString(), anyString())).thenReturn(new java.util.Properties())
 
-    val oldConfig =  KafkaConfig.fromProps(TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect, port = 9092))
+    val initialProps = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect, port = 9092)
```
Comment (jolshan) [inline core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala:612]: Can you explain this change?

Comment (splett2) [reply core/src/test/scala/unit/kafka/server/DynamicBrokerConfigTest.scala:612]: The `TestDynamicThreadPool` mock class expects the various thread count props to be initialized to the defaults. So I am removing the static overrides.

**Review by jolshan — APPROVED** [review]

**Comment by splett2** [comment]
@jolshan 
I am investigating
```
[kafka.api.SslAdminIntegrationTest.testAsynchronousAuthorizerAclUpdatesDontBlockRequestThreads()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13655/4/testReport/junit/kafka.api/SslAdminIntegrationTest/Build___JDK_11_and_Scala_2_13___testAsynchronousAuthorizerAclUpdatesDontBlockRequestThreads__/)
```

I will revert the IO threads thing for now until I investigate further.

**Pull request merged**